src/main/java/org/traccar/protocol/RuptelaProtocolDecoder.java

```diff
@@ -236,7 +237,13 @@ protected Object decode(
                 buf.readUnsignedByte(); // timestamp extension
 
                 if (type == MSG_EXTENDED_RECORDS) {
-                    buf.readUnsignedByte(); // record extension
+                    int recordExtension = buf.readUnsignedByte();
+                    int mergeRecordCount = BitUtil.between(recordExtension, 4, 8);
```

Comment:
One last minor thing
```suggestion
                    int mergeRecordCount = BitUtil.from(recordExtension, 4);
```